packages/catalog-server/src/lib/catalog.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {getDistTagsForVersion} from './npm.js';
+import {validatePackage} from '@webcomponents/custom-elements-manifest-tools/lib/validate.js';
+import {getCustomElements} from '@webcomponents/custom-elements-manifest-tools';
+import {PackageFiles} from '@webcomponents/custom-elements-manifest-tools/lib/npm.js';
+import {Repository} from './repository.js';
+import {PackageVersion} from '@webcomponents/catalog-api/lib/schema.js';
+
+export interface CatalogInit {
+  repository: Repository;
+  files: PackageFiles;
+}
+
+/**
+ * Implements operations for reading and writing to a catalog.
+ */
+export class Catalog {
+  private repository: Repository;
```

Comment:
We can use real #private fields

Reply:
Ok, as long as we have a target of es2022 or newer. I don't want to compile to WeakMaps.

Reply:
Set target to es2022 and changed to real private fields.